lib/logger/lib/logger/formatter.ex

```diff
@@ -121,9 +121,16 @@ defmodule Logger.Formatter do
   defp metadata(pid) when is_pid(pid) do
     :erlang.pid_to_list(pid)
   end
-  defp metadata(pid) when is_reference(pid) do
-    '#Ref' ++ rest = :erlang.ref_to_list(pid)
+  defp metadata(ref) when is_reference(ref) do
+    '#Ref' ++ rest = :erlang.ref_to_list(ref)
     rest
   end
+  defp metadata(atom) when is_atom(atom) do
+    case Atom.to_string(atom) do
+      "Elixir." <> rest -> rest
+      "nil" -> ""
```

Comment:
Yup. This code is perfect as is. Let's just hand the nil application case,
which can be in this PR or another one. Let me know.

-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*